### ISSUE TITLE
False positive with Style/StringMethods cop

### ISSUE THREAD

**Issue by tfausak** [bbatsov/rubocop#2260]
The Style/StringMethods cop appears to be a little overzealous when recommending `to_sym` over `intern`.

``` rb
# Gemfile
source 'https://rubygems.org'
gem 'rubocop',
    github: 'bbatsov/rubocop',
    ref: 'c974769e9021949d545d7484490536d707a38cf1'
```

``` rb
# example.rb
C = Class.new { def self.intern; end }
C.intern
```

``` sh
> bundle exec rubocop -V
0.34.1 (using Parser 2.2.2.6, running on ruby 2.2.2 x86_64-darwin14)
> bundle exec rubocop -D example.rb 
Inspecting 1 file
C

Offenses:

example.rb:2:3: C: Style/StringMethods: Prefer to_sym over intern.
C.intern
  ^^^^^^

1 file inspected, 1 offense detected
```

**Comment by bbatsov** [comment]
Probably we should disable this cop by default. I was under the impression that only `String` had this method.

**Comment by tfausak** [comment]
I think that's true in the standard library, but there are [many classes that define `intern`](https://www.omniref.com/?q=#intern&language=ruby).

**Comment by tfausak** [comment]
:+1:
